packages/react/select/src/Select.tsx

```diff
@@ -306,34 +293,83 @@ SelectIcon.displayName = ICON_NAME;
 
 const CONTENT_NAME = 'SelectContent';
 
+type ReactOption = React.ReactElement<React.ComponentProps<'option'>>;
+
+type SelectNativeOptionsContextValue = {
+  onNativeOptionAdd(option: ReactOption): void;
+  onNativeOptionRemove(option: ReactOption): void;
+};
+const [SelectNativeOptionsProvider, useSelectNativeOptionsContext] =
+  createSelectContext<SelectNativeOptionsContextValue>(CONTENT_NAME);
+
 type SelectContentElement = SelectContentImplElement;
 interface SelectContentProps extends SelectContentImplProps {}
 
 const SelectContent = React.forwardRef<SelectContentElement, SelectContentProps>(
   (props: ScopedProps<SelectContentProps>, forwardedRef) => {
     const context = useSelectContext(CONTENT_NAME, props.__scopeSelect);
     const [fragment, setFragment] = React.useState<DocumentFragment>();
+    // We set this to true by default so that events bubble to forms without JS (SSR)
+    const isFormControl = context.trigger ? Boolean(context.trigger.closest('form')) : true;
+    const [nativeOptionsSet, setNativeOptionsSet] = React.useState(new Set<ReactOption>());
+
+    const nativeSelectKey = Array.from(nativeOptionsSet.values())
+      .map((option) => option.props.value)
+      .join(';');
```

Comment:
We should try to explain why we need to do this, something like this? (probably bad first attempt):

```
In order for the native select value to update correctly, we need to track and re-render the entire element whenever it's options are changed
```